### PR TITLE
Index metadata_json for each mint

### PR DESCRIPTION
**Pull request by imabdulbasit**

### CODE REVIEW

File: api/src/entities/collection_mints.rs [hunk 1]
```diff
@@ -82,6 +94,8 @@ pub enum Relation {
         on_delete = "Cascade"
     )]
     Collections,
+    #[sea_orm(has_many = "super::metadata_jsons::Entity")]
```
Comment (kespinola) [inline api/src/entities/collection_mints.rs:97]: This should be a has_one, yes?

File: api/src/metadata_json.rs [hunk 2]
```diff
@@ -14,22 +14,54 @@ use crate::{
...
+    /// # Arguments
+    ///
+    /// * `id` - The ID of the metadata to fetch.
+    /// * `db` - The database connection to use.
+    ///
+    /// # Errors
+    ///
+    /// This function fails if there is no matching `metadata_json` entry in the database
+    /// or if it is unable to fetch related data from the database
+    pub async fn fetch(id: Uuid, db: &Connection) -> Result<Self> {
```
Comment (kespinola) [inline api/src/metadata_json.rs:42]: Nice. Good job on docs too. We are going to publish cargo docs soon for our services.

File: api/src/metadata_json.rs [hunk 3]
```diff
@@ -63,7 +94,7 @@ impl MetadataJson {
             .ok_or_else(|| anyhow!("no uri. call #upload before #save"))?;
 
         let metadata_json_active_model = metadata_jsons::ActiveModel {
-            collection_id: Set(collection),
+            id: Set(id),
```
Comment (kespinola) [inline api/src/metadata_json.rs:97]: Do we have migration to move collection_id to id? Haven't gotten to the migrations yet.

File: api/src/mutations/mint.rs [hunk 4]
```diff
@@ -136,8 +137,14 @@ impl Mutation {
         };
 
         let collection_mint_model = collection_mint_active_model.insert(conn).await?;
+
         let proto_blockchain_enum: proto::Blockchain = collection.blockchain.into();
 
+        MetadataJson::fetch(collection.id, db)
+            .await?
+            .save(collection_mint_model.id, db)
+            .await?;
```
Comment (kespinola) [inline api/src/mutations/mint.rs:146]: Nice job.

File: migration/src/m20230405_133333_rename_collection_id_to_id_and_rm_fk.rs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                    )
+                    .to_owned(),
+            )
+            .await?;
+
+        manager
+            .alter_table(
+                Table::alter()
+                    .table(MetadataJsons::Table)
+                    .rename_column(Alias::new("collection_id"), Alias::new("id"))
```
Comment (kespinola) [inline migration/src/m20230405_133333_rename_collection_id_to_id_and_rm_fk.rs:120]: Ok rename here 👍

File: api/src/entities/collection_mints.rs (hunk 1 above)
Comment (imabdulbasit) [reply api/src/entities/collection_mints.rs:97]: yeah , sea-orm-cli generates 1:M relation for every entity by default

Comment (imabdulbasit) [reply api/src/entities/collection_mints.rs:97]: yeah , sea-orm-cli generates 1:M relation for every entity by default

**Pull request merged**